### PR TITLE
server: move SET CLUSTER SETTINGS tests to unit tests

### PR DESCRIPTION
**Pull request by dt**
doing this in logic tests can be flaky since logic tests are run in parallel.

### CODE REVIEW

**Comment by RaduBerinde** [comment]
LGTM, thanks!

**Comment by arjunravinarayan** [comment]
Pull the enum tests out too?

**Comment by arjunravinarayan** [comment]
Could add set/show enums as well, or I can add it in a separate PR; your call.

LGTM.

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15461#-:-KionopjSC2iF2g9J50p:bu83qlo)*

**Comment by dt** [comment]
Rebased over the enum conflicts, RFAL.

**Comment by arjunravinarayan** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15461#-:-KioqcAGTRumblIWV6z7:bnfp4nl)*

**Comment by RaduBerinde** [comment]
LGTM

File: pkg/server/settingsworker_test.go [hunk 1]
```diff
@@ -34,11 +36,13 @@ const strKey = "testing.str"
 const intKey = "testing.int"
 const durationKey = "testing.duration"
 const byteSizeKey = "testing.bytesize"
+const enumKey = "testing.enum"
 
 var strA = settings.RegisterStringSetting(strKey, "", "<default>")
 var intA = settings.RegisterIntSetting(intKey, "", 1)
 var durationA = settings.RegisterDurationSetting(durationKey, "", time.Minute)
 var byteSizeA = settings.RegisterByteSizeSetting(byteSizeKey, "", 1024*1024)
+var enumA = settings.RegisterEnumSetting(enumKey, "", "foo", map[int64]string{1: "foo", 2: "bar"})
```
Comment (RaduBerinde) [inline pkg/server/settingsworker_test.go]: s/enumA// (style check failure)

Comment (dt) [inline pkg/server/settingsworker_test.go]: added the call below to match others.

Comment (RaduBerinde) [inline pkg/server/settingsworker_test.go]: looks good

**Pull request merged**